python/cudf/cudf/_lib/column.pyx

```diff
@@ -497,6 +548,26 @@ cdef class Column:
             children=tuple(children)
         )
 
+    @staticmethod
+    def from_pylibcudf_column(
+        pylibcudf.Column col, bint data_ptr_exposed=False
+    ):
+        # TODO: Rewrite utility for dtype conversion to not need a column view.
+        dtype = dtype_from_column_view(dereference(col.get_underlying()))
+
+        return cudf.core.column.build_column(
+            data=as_buffer(col.data.obj) if col.data is not None else None,
+            dtype=dtype,
+            size=col.size,
+            mask=as_buffer(col.mask.obj) if col.mask is not None else None,
+            offset=col.offset,
+            null_count=col.null_count,
+            children=tuple([
+                Column.from_pylibcudf_column(child)
+                for child in col.children
+            ])
+        )
```

Comment:
What is your longer-term vision for how this "should" work? To my mind it seems like what we _want_ is for cudf python-exposed columns to own a pylibcudf column and just layer syntax sugar and (perhaps) pandas-specific behaviour on top.

I guess the challenge is how to do this migration piecemeal.

Reply:
Yes, that is exactly how I've been envisioning it. There's some more thought that will have to go into it since there's also cases where we might want to cache the pylibcudf Table representing a cudf Frame for calling functions on a libcudf::table, but we need a single source of truth for the ownership pathway (i.e. it makes more sense for each cudf  column to own a pylibcudf column than for each cudf Frame to own a list of pylibcudf columns due to the fact that we need to store additional metadata like dtypes per column because there isn't a 1-1 mapping between Python and C++ dtypes). We'll definitely have to spend some time getting that mapping right once we've done enough the migration. For now, though, I'm OK eating a bit of overhead to keep the migration simple and as minimally intrusive as possible as demonstrated in this PR.